docs/guides/operator-private-gcp-confidential-space.md

```diff
@@ -0,0 +1,405 @@
+
+In this section:
+- [Install Terraform](#install-terraform)
+- [Set Up the Environment](#set-up-the-environment)
+- [Run Terraform](#run-terraform)
+- [Test](#test)
+- [Clean Up](#clean-up)
+- [Inputs](#inputs)
+- [Outputs](#outputs)
+- [Terraform Template&#8212;Notes](#terraform-templatenotes)
+
+#### Install Terraform
+
+Install Terraform if it is not already installed: visit [terraform.io](https://www.terraform.io/).
+
+#### Set Up the Environment
+
+1. Set the project, replacing the `{PROJECT_ID}` placeholder with your own project ID (see [Confidential Space Account Setup](#confidential-space-account-setup)):
+
+   ```
+   gcloud config set project {PROJECT_ID}
+   ```
+
+2. Configure the environment for Terraform:
+
+   ```
+   gcloud auth application-default login
+   ```
+
+#### Run Terraform
+
+Run the following:
+
+```
+terraform init
+terraform apply
+```
+
+#### Test
+
+1. Get the public IP address for the load balancer:
+
+   ```
+   terraform output load-balancer-ip
+   ```
+
+2. Check health at the health check endpoint (for details, see [Test Using the Health Check Endpoint](#test-using-the-health-check-endpoint)):
+
+   ```
+   http://{PUBLIC_IP}/ops/healthcheck
+   ```
+
+#### Clean Up
+
+Remove all resources created by Terraform:
+
+```
+terraform destroy
+```
+
+#### Inputs
+
+(**GWH_YS18 I need more data re the inputs and outputs so we can have some little intro line to these tables. Is this something they clean up? Or something that's required? Please help with some explanation for both inputs and outputs.**)
+
+| Name               | Description | Type     | Default             | Required |
+|--------------------|-------------|----------|---------------------|:--------:|
+| project_id         | n/a         | `string` | n/a                 |   yes    |
+| service_account    | n/a         | `string` | n/a                 |   yes    |
+| uid_operator_image | n/a         | `string` | n/a                 |   yes    |
+| uid_api_token      | n/a         | `string` | n/a                 |   yes    |
+| region             | n/a         | `string` | `"asia-southeast1"` |    no    |
+| network_name       | n/a         | `string` | `"uid-operator"`    |    no    |
+| uid_machine_type   | n/a         | `string` | `"n2d-standard-16"` |    no    |
+| uid_deployment_env | n/a         | `string` | `"integ"`           |    no    |
+| max_replicas       | n/a         | `number` | `5`                 |    no    |
+| min_replicas       | n/a         | `number` | `1`                 |    no    |
+| debug_mode         | n/a         | `bool`   | `false`             |    no    |
+
+#### Outputs
+
+| Name             | Description |
+|------------------|-------------|
+| load-balancer-ip | n/a         |
+
+#### Terraform Template&#8212;Notes
+
+1. You might want to change the load balancer from HTTP to HTTPS. To do this, follow these steps: (**GWH_YS19 surely they would have to use HTTPS? Should this be higher profile than a note at the end?**)
```

Comment:
Our recommend deployment way, as Andrei mentioned, is to deploy with LB (scaling group) + https enabled.

It's our recommendation, but not enforced.